### ISSUE TITLE
Creating symbolic links is broken when paths are virtualized

### ISSUE THREAD

**Issue by lostmsu** [dotnet/runtime#91044]
### Description

My app is packaged using DesktopBridge to be published in Store.

The issue is that when I create a symbolic link from withing my app's subfolder of `AppData\Local` (`Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData) + "\\MyApp"`), it is not functional as it points to non-virtualized location.

### Reproduction Steps

This has to be run from a desktop bridge app
```csharp
bool virtualized = new DesktopBridge.Helpers().IsRunningAsUwp();
Console.WriteLine($"virtualized: {virtualized}");

var localData = new DirectoryInfo(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData));
var appData = new DirectoryInfo(Path.Join(localData.FullName, "SymlinkTest"));
appData.Create();


string original = Path.Join(appData.FullName, "original.txt");
File.WriteAllText(path: original, contents: "42");
var symlink = new FileInfo(Path.Join(appData.FullName, "symlink.txt"));
if (symlink.Exists) {
    symlink.Delete();
}
symlink.CreateAsSymbolicLink(original);
Console.WriteLine(File.ReadAllText(symlink.FullName));
symlink.Delete();
```

### Expected behavior

> virtualized: true
> 42

### Actual behavior

> virtualized: true
> System.IO.DirectoryNotFoundException: 'Could not find a part of the path 'C:\Users\MYUSER\AppData\Local\SymlinkTest\symlink.txt'.'

**Comment by adamsitnik** [comment]
@Jozkee could you PTAL?

**Comment by Jozkee** [comment]
Is `CreateAsSymbolicLink` silently failing?
@lostmsu do you have steps on how to run that code on a Desktop bridge app?

**Comment by lostmsu** [comment]
> silently failing?
No, it creates a symlink, but the destination is non-virtualized path, and trying to read the content immediately fails. So does an attempt to remove the symlink.

See the attached project sample.

If it was not clear - Windows only.

[SymlinkPackaged.zip](https://github.com/dotnet/runtime/files/12433483/SymlinkPackaged.zip)

**Comment by Jozkee** [comment]
> No, it creates a symlink, but the destination is non-virtualized path, and trying to read the content immediately fails. So does an attempt to remove the symlink.

I see, yes. I think this may be a problem related to the windows API CreateSymbolicLinkW which doesn't take into account the virtualized context, so we may not be able to fix it.

**Comment by bansalsushant** [comment]
@lostmsu Which environment are you using for this?
We tried to repro this on Win 11 23H2 and we were not able to repro the issue - the sym link worked as expected.
